iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/env/RemoveDataNodeManager.java

```diff
@@ -0,0 +1,457 @@
+  }
+
+  /**
+   * Broadcasts DataNodes' status change, preventing disabled DataNodes from accepting read or write
+   * requests.
+   *
+   * @param dataNodes the list of DataNodes that require broadcast status changes
+   */
+  public void broadcastDataNodeStatusChange(List<TDataNodeLocation> dataNodes) {
+    String dataNodesString =
+        dataNodes.stream()
+            .map(RegionMaintainHandler::getIdWithRpcEndpoint)
+            .collect(Collectors.joining(", "));
+    LOGGER.info(
+        "{}, BroadcastDataNodeStatusChange start, dataNode: {}",
+        REMOVE_DATANODE_PROCESS,
+        dataNodesString);
+
+    List<TDataNodeConfiguration> otherOnlineDataNodes =
+        configManager.getNodeManager().filterDataNodeThroughStatus(NodeStatus.Running).stream()
+            .filter(node -> !dataNodes.contains(node.getLocation()))
+            .collect(Collectors.toList());
+
+    for (TDataNodeConfiguration node : otherOnlineDataNodes) {
+      TCleanDataNodeCacheReq disableReq = new TCleanDataNodeCacheReq(dataNodes);
+      TSStatus status =
+          (TSStatus)
+              SyncDataNodeClientPool.getInstance()
+                  .sendSyncRequestToDataNodeWithRetry(
+                      node.getLocation().getInternalEndPoint(),
+                      disableReq,
+                      CnToDnRequestType.CLEAN_DATA_NODE_CACHE);
+      if (!isSucceed(status)) {
+        LOGGER.error(
+            "{}, BroadcastDataNodeStatusChange meets error, dataNode: {}, error: {}",
+            REMOVE_DATANODE_PROCESS,
+            dataNodesString,
+            status);
+        return;
+      }
+    }
+
+    LOGGER.info(
+        "{}, BroadcastDataNodeStatusChange finished, dataNode: {}",
+        REMOVE_DATANODE_PROCESS,
+        dataNodesString);
+  }
+
+  /**
+   * Removes a batch of DataNodes from the node information.
+   *
+   * @param removedDataNodes the list of DataNodeLocations to be removed
+   */
+  public void removeDataNodePersistence(List<TDataNodeLocation> removedDataNodes) {
+    // Remove consensus record
+    try {
+      configManager.getConsensusManager().write(new RemoveDataNodePlan(removedDataNodes));
+    } catch (ConsensusException e) {
+      LOGGER.warn("Failed in the write API executing the consensus layer due to: ", e);
+    }
+
+    // Adjust maxRegionGroupNum
+    configManager.getClusterSchemaManager().adjustMaxRegionGroupNum();
+
+    // Remove metrics
+    for (TDataNodeLocation dataNodeLocation : removedDataNodes) {
+      PartitionMetrics.unbindDataNodePartitionMetricsWhenUpdate(
+          MetricService.getInstance(),
+          NodeUrlUtils.convertTEndPointUrl(dataNodeLocation.getClientRpcEndPoint()));
+    }
+  }
+
+  /**
+   * Stops the specified old DataNodes.
+   *
+   * @param removedDataNodes the list of DataNodeLocations to be stopped
+   */
+  public void stopDataNodes(List<TDataNodeLocation> removedDataNodes) {
+    removedDataNodes.parallelStream().forEach(this::stopDataNode);
```

Comment:
Please use asynchronous parallel way, do not use parallelStream

Reply:
refactored